### PR TITLE
Raise pulpcore requirement to 3.55+

### LINKED ISSUE #1100: Make plugin compatible with with pulpcore 3.55+

### PR DESCRIPTION
**Pull request by hstct**
closes #1100

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # All things django and asyncio are deliberately left to pulpcore
 # Example transitive requirements: asgiref, asyncio, aiohttp
-pulpcore>=3.49.0,<3.55
+pulpcore>=3.55.0,<3.70
```
Comment (quba42) [inline requirements.txt:3]: I don't think we need to raise the floor, just the ceiling?
```suggestion
pulpcore>=3.49.0,<3.70
```

File: CHANGES/1100.feature [hunk 2]
```diff
@@ -0,0 +1 @@
+Made the plugin compatible with pulpcore 3.55.0+.
```
Comment (quba42) [inline CHANGES/1100.feature:1]: I had a look at other plugins, and the formulations for this are all over the place.
Let's use this formulation this time (consistent with our last upper bound bump), but maybe start a followup discussion in the Pulp discourse.

**Comment by mdellweg** [comment]
Oh, I think you are in for a search and replace session.
e.g. `artifacts_api_client` is now `pulpcore_bindings.ArtifactsApi`.

If used this scriptlet:
```bash
find . -name "*.py" -exec sed -i \
-e 's/content_api_client/pulpcore_bindings.ContentApi/' \
-e 's/artifacts_api_client/pulpcore_bindings.ArtifactsApi/' \
-e 's/upstream_pulp_api_client/pulpcore_bindings.UpstreamPulpsApi/' \
\{\} +
```

But you need to adjust the fixture calling in the function headers by hand.

**Review by quba42 — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (quba42) [reply requirements.txt:3]: pulp_rpm just raised the upper bound as well.

File: CHANGES/1100.feature (hunk 2 above)
Comment (hstct) [reply CHANGES/1100.feature:1]: I just copied it from the last time we did this.

Comment (quba42) [reply CHANGES/1100.feature:1]: I think that is the best we can do for now. I added a comment on Discourse about this: https://discourse.pulpproject.org/t/pulpcore-3-55-is-coming-soon/1259

Comment (mdellweg) [reply CHANGES/1100.feature:1]: If you want it consistently, you need to automate it.

**Review by quba42 — COMMENTED** [review]
I am good with the changes as soon as the tests turn green.

**Review by quba42 — APPROVED** [review]

**Pull request merged**